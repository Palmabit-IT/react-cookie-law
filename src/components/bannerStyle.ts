const bannerStyle = {
  dialog: {
    position: 'fixed',
    top: 0,
    left: 0,
    right: 0,
    zIndex: 100000,
    backgroundColor: '#f8f7f7',
    padding: '10px',
  },
  container: {
    maxWidth: '960px',
    marginLeft: 'auto',
    marginRight: 'auto',
    overflow: 'hidden',
  },
  message: {
    minHeight: '32px',
    fontSize: '10pt',
    fontWeight: 400,
    lineHeight: '130%',
    padding: '10px 0',
    color: '#000000',
  },
  policy: {
    fontSize: '10pt',
    marginLeft: '10px',
    color: '#000000',
    textDecoration: 'underline',
  },
  selectPane: {
    display: 'table-cell',
    padding: '3px 0px 10px 0px',
  },
  optionWrapper: {
    display: 'inline-block',
    position: 'relative',
    marginRight: '7px',
    marginLeft: '7px',
    float: 'left',
  },
  optionLabel: {
    height: 'auto',
    width: 'auto',
    minHeight: '14px',
    fontSize: '12pt',
    color: '#c6c6c6',
    display: 'inline-block',
    padding: '1px 0 0 20px',
    position: 'relative',
    top: 0,
    left: 0,
    zIndex: 1,
    cursor: 'default',
    verticalAlign: 'top',
  },
  checkbox: {
    position: 'absolute',
    top: '4px',
    left: 0,
    width: '14px',
    height: '14px',
    zIndex: 2,
    cursor: 'pointer',
  },
  buttonWrapper: {
    float: 'right',
  },
  button: {
    display: 'inline-block',
    backgroundColor: '#000000',
    padding: '3px',
    minWidth: '80px',
    color: '#ffffff',
    textDecoration: 'none',
    fontSize: '10pt',
    fontWeight: 400,
    marginRight: '5px',
    marginLeft: '5px',
    textAlign: 'center',
    whiteSpace: 'nowrap',
    cursor: 'pointer',
  },
} as const;

export type StyleableComponents = keyof typeof bannerStyle

export default bannerStyle
