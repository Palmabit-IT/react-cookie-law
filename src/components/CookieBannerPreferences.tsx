import React from 'react';
import CookieOption from './CookieOption';
import bannerStyle, { StyleableComponents } from './bannerStyle';

import type { CoryphaPreference } from '../coryphaAPI';

export interface Props {
  styles?: Partial<Record<StyleableComponents, React.CSSProperties>>
  necessaryOptionText?: string
  preferencesOptionText?: string
  statisticsOptionText?: string
  marketingOptionText?: string
  showPreferencesOption?: boolean
  showStatisticsOption?: boolean
  showMarketingOption?: boolean
  preferencesDefaultChecked?: boolean
  statisticsDefaultChecked?: boolean
  marketingDefaultChecked?: boolean
  onTogglePreferencesCookies?: (checked?: boolean) => void
  onToggleStatisticsCookies?: (checked?: boolean) => void
  onToggleMarketingCookies?: (checked?: boolean) => void
  coryphaPreferences?: CoryphaPreference[]
  onToggleCoryphaPreference?: (active: boolean, preference: CoryphaPreference) => void
}
const CookieBannerPreferences: React.FC<Props> = (props: Props = {}) => {
  const {
    styles = {},
    necessaryOptionText = 'Necessary',
    preferencesOptionText = 'Preferences',
    statisticsOptionText = 'Statistics',
    marketingOptionText = 'Marketing',
    showPreferencesOption = true,
    showStatisticsOption = true,
    showMarketingOption = true,
    preferencesDefaultChecked = false,
    statisticsDefaultChecked = false,
    marketingDefaultChecked = false,
    onTogglePreferencesCookies = () => undefined,
    onToggleStatisticsCookies = () => undefined,
    onToggleMarketingCookies = () => undefined,
    coryphaPreferences = [],
    onToggleCoryphaPreference = () => undefined,
  } = props;

  const {
    selectPane: selectPaneStyle,
    optionWrapper: optionWrapperStyle,
    optionLabel: optionLabelStyle,
    checkbox: checkboxStyle,
  } = { ...bannerStyle, ...styles };

  const cookieOptionStyle = {
    optionWrapperStyle,
    optionLabelStyle,
    checkboxStyle,
  };

  return (
    <div className="react-cookie-law-select-pane" style={selectPaneStyle}>
      {coryphaPreferences.length > 0 ? (
        <>
          <CookieOption
            id="check-required-cookies"
            text={necessaryOptionText}
            styles={cookieOptionStyle}
            disabled
            checked
          />

          {coryphaPreferences.map((preference) => (
            <CookieOption
              key={preference.id}
              id={preference.id}
              text={preference.name}
              styles={cookieOptionStyle}
              disabled={preference.required}
              checked={preference.required}
              onChange={(value) => onToggleCoryphaPreference(value, preference)}
            />
          ))}
        </>
      ) : (
        <>
          <CookieOption
            id="check-required-cookies"
            text={necessaryOptionText}
            styles={cookieOptionStyle}
            disabled
            checked
          />

          {showPreferencesOption && (
            <CookieOption
              id="check-preferences-cookies"
              text={preferencesOptionText}
              styles={cookieOptionStyle}
              checked={preferencesDefaultChecked}
              onChange={onTogglePreferencesCookies}
            />
          )}

          {showStatisticsOption && (
            <CookieOption
              id="check-statistics-cookies"
              text={statisticsOptionText}
              styles={cookieOptionStyle}
              checked={statisticsDefaultChecked}
              onChange={onToggleStatisticsCookies}
            />
          )}

          {showMarketingOption && (
            <CookieOption
              id="check-marketing-cookies"
              text={marketingOptionText}
              styles={cookieOptionStyle}
              checked={marketingDefaultChecked}
              onChange={onToggleMarketingCookies}
            />
          )}
        </>
      )}
    </div>
  );
};

export default CookieBannerPreferences
