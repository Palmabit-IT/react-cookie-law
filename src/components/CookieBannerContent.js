import React from 'react';
import bannerStyle from './bannerStyle';
import CookieBannerPreferences from './CookieBannerPreferences';

class CookieBannerContent extends React.Component {
  constructor(props) {
    super(props);

    this.state = { showPreferences: false };
  }

  render() {
    const {
      styles = {},
      className = '',
      message = 'No text',
      policyLink = '/#',
      privacyPolicyLinkText = 'Privacy Policy',
      acceptButtonText = 'Accept all',
      managePreferencesButtonText = 'Mange my cookies',
      savePreferencesButtonText = 'Save and close',
      onConfirm = () => {},
      onAcceptAll = () => {},
    } = this.props;

    const { showPreferences } = this.state;

    const {
      dialog: dialogStyle,
      container: containerStyle,
      message: messageStyle,
      policy: policyStyle,
      buttonWrapper: buttonWrapperStyle,
      button: buttonStyle,
    } = { ...bannerStyle, ...styles };

    return (
      <div
        className={`react-cookie-law-dialog ${className}`}
        style={dialogStyle}
      >
        <div className="react-cookie-law-container" style={containerStyle}>
          <div className="react-cookie-law-msg" style={messageStyle}>
            {message}
          </div>

          {showPreferences && <CookieBannerPreferences {...this.props} />}

          <a
            href={policyLink}
            target="_blank"
            className="react-cookie-law-policy"
            style={policyStyle}
          >
            {privacyPolicyLinkText}
          </a>

          <div
            className="react-cookie-law-button-wrapper"
            style={buttonWrapperStyle}
          >
            {showPreferences ? (
              <button
                type="button"
                className="react-cookie-law-save-btn"
                style={buttonStyle}
                onClick={() => onConfirm()}
              >
                <span>{savePreferencesButtonText}</span>
              </button>
            ) : (
              <button
                type="button"
                className="react-cookie-law-manage-btn"
                style={buttonStyle}
                onClick={() => this.setState({ showPreferences: true })}
              >
                <span>{managePreferencesButtonText}</span>
              </button>
            )}
            <button
              type="button"
              className="react-cookie-law-accept-btn"
              style={buttonStyle}
              onClick={() => onAcceptAll()}
            >
              <span>{acceptButtonText}</span>
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default CookieBannerContent;
