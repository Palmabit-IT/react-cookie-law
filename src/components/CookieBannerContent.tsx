import React from 'react';
import bannerStyle from './bannerStyle';
import CookieBannerPreferences from './CookieBannerPreferences';

interface Props extends React.ComponentProps<typeof CookieBannerPreferences> {
  acceptButtonText?: string
  className?: string
  // Not used to-date.
  declineButtonText?: string
  managePreferencesButtonText?: string
  message?: string
  onAcceptAll?: () => void
  // Not used to-date.
  onDecline?: () => void
  onConfirm?: () => void
  policyLink?: string
  privacyPolicyLinkText?: string
  savePreferencesButtonText?: string
  // Not used to-date.
  showDeclineButton?: boolean
}

interface State {
  showPreferences?: boolean
}
class CookieBannerContent extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { showPreferences: false };
  }

  render() {
    const {
      acceptButtonText = 'Accept all',
      className = '',
      managePreferencesButtonText = 'Mange my cookies',
      message = 'No text',
      onAcceptAll = () => undefined,
      onConfirm = () => undefined,
      policyLink = '/#',
      privacyPolicyLinkText = 'Privacy Policy',
      savePreferencesButtonText = 'Save and close',
      ...preferenceProps
    } = this.props;

    const { showPreferences } = this.state;

    const {
      dialog: dialogStyle,
      container: containerStyle,
      message: messageStyle,
      policy: policyStyle,
      buttonWrapper: buttonWrapperStyle,
      button: buttonStyle,
    } = { ...bannerStyle, ...preferenceProps.styles };

    return (
      <div
        className={`react-cookie-law-dialog ${className}`}
        style={dialogStyle}
      >
        <div className="react-cookie-law-container" style={containerStyle}>
          <div className="react-cookie-law-msg" style={messageStyle}>
            {message}
          </div>

          {showPreferences && <CookieBannerPreferences {...preferenceProps} />}

          <a
            href={policyLink}
            className="react-cookie-law-policy"
            style={policyStyle}
          >
            {privacyPolicyLinkText}
          </a>

          <div
            className="react-cookie-law-button-wrapper"
            style={buttonWrapperStyle}
          >
            {showPreferences ? (
              <button
                type="button"
                className="react-cookie-law-save-btn"
                style={buttonStyle}
                onClick={() => onConfirm()}
              >
                <span>{savePreferencesButtonText}</span>
              </button>
            ) : (
              <button
                type="button"
                className="react-cookie-law-manage-btn"
                style={buttonStyle}
                onClick={() => this.setState({ showPreferences: true })}
              >
                <span>{managePreferencesButtonText}</span>
              </button>
            )}
            <button
              type="button"
              className="react-cookie-law-accept-btn"
              style={buttonStyle}
              onClick={() => onAcceptAll()}
            >
              <span>{acceptButtonText}</span>
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default CookieBannerContent;
