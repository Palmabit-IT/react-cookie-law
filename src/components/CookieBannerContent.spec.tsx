import React from 'react';
import { mount, shallow } from 'enzyme';
import CookieBannerContent from './CookieBannerContent';
import CookieBannerPreferences from './CookieBannerPreferences';
import bannerStyle from './bannerStyle';

describe('CookieBannerContent component', () => {
  test('should be rendered', () => {
    const component = shallow(
      <CookieBannerContent message="Custom text" />,
    );

    expect(component).toMatchSnapshot();
  });

  test('should be rendered with elements', () => {
    const props = {
      styles: bannerStyle,
      message: 'Custom text',
      policyLink: '/url-to-policy',
      privacyPolicyLinkText: 'Privacy Policy',
      necessaryOptionText: 'Necessary',
      preferencesOptionText: 'Preferences',
      statisticsOptionText: 'Statistics',
      marketingOptionText: 'Marketing',
      acceptButtonText: 'Accept all',
      managePreferencesButtonText: 'Manage my cookies',
      savePreferencesButtonText: 'Save and close',
    };

    const component = mount(
      <CookieBannerContent {...props} />,
    );

    expect(component.find('.react-cookie-law-msg').text()).toBe('Custom text');
    expect(component.find('a.react-cookie-law-policy').prop('href')).toBe('/url-to-policy');
    expect(component.find('.react-cookie-law-policy').text()).toBe('Privacy Policy');
    expect(component.contains(<CookieBannerPreferences {...props} />)).toBeFalsy();
  });

  test('should click manage preferences button', () => {
    const props = {
      onConfirm: jest.fn(),
    };

    const component = mount(
      <CookieBannerContent {...props} />,
    );

    component.find('.react-cookie-law-manage-btn').simulate('click');

    expect(component.contains(<CookieBannerPreferences />)).toBeTruthy();
  });

  test('should click confirm button', () => {
    const props = {
      onAcceptAll: jest.fn(),
    };

    const component = mount(
      <CookieBannerContent {...props} />,
    );

    component.find('.react-cookie-law-accept-btn').simulate('click');

    expect(props.onAcceptAll).toHaveBeenCalledTimes(1);
  });
});
