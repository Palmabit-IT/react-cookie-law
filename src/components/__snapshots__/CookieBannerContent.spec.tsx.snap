// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieBannerContent component should be rendered 1`] = `
<div
  className="react-cookie-law-dialog "
  style={
    Object {
      "backgroundColor": "#f8f7f7",
      "left": 0,
      "padding": "10px",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "zIndex": 100000,
    }
  }
>
  <div
    className="react-cookie-law-container"
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": "960px",
        "overflow": "hidden",
      }
    }
  >
    <div
      className="react-cookie-law-msg"
      style={
        Object {
          "color": "#000000",
          "fontSize": "10pt",
          "fontWeight": 400,
          "lineHeight": "130%",
          "minHeight": "32px",
          "padding": "10px 0",
        }
      }
    >
      Custom text
    </div>
    <a
      className="react-cookie-law-policy"
      href="/#"
      style={
        Object {
          "color": "#000000",
          "fontSize": "10pt",
          "marginLeft": "10px",
          "textDecoration": "underline",
        }
      }
    >
      Privacy Policy
    </a>
    <div
      className="react-cookie-law-button-wrapper"
      style={
        Object {
          "float": "right",
        }
      }
    >
      <button
        className="react-cookie-law-manage-btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#000000",
            "color": "#ffffff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "10pt",
            "fontWeight": 400,
            "marginLeft": "5px",
            "marginRight": "5px",
            "minWidth": "80px",
            "padding": "3px",
            "textAlign": "center",
            "textDecoration": "none",
            "whiteSpace": "nowrap",
          }
        }
        type="button"
      >
        <span>
          Mange my cookies
        </span>
      </button>
      <button
        className="react-cookie-law-accept-btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#000000",
            "color": "#ffffff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": "10pt",
            "fontWeight": 400,
            "marginLeft": "5px",
            "marginRight": "5px",
            "minWidth": "80px",
            "padding": "3px",
            "textAlign": "center",
            "textDecoration": "none",
            "whiteSpace": "nowrap",
          }
        }
        type="button"
      >
        <span>
          Accept all
        </span>
      </button>
    </div>
  </div>
</div>
`;
